packages/terminal/src/browser/terminal-widget.ts

```diff
@@ -72,41 +77,46 @@ export class TerminalWidget extends BaseWidget {
     }
 
     protected registerResize(): void {
-        let initialGeometry = (this.term as any).proposeGeometry()
+        const initialGeometry = (this.term as any).proposeGeometry()
         this.cols = initialGeometry.cols;
         this.rows = initialGeometry.rows;
 
         this.term.on('resize', size => {
-            if (!this.pid) {
+            if (!this.terminalId) {
                 return;
             }
             this.cols = size.cols
             this.rows = size.rows
-            let url = this.endpoint.getRestUrl().toString() + "/" + this.pid + '/size?cols=' + this.cols + '&rows=' + this.rows;
-            fetch(url, { method: 'POST' })
+            this.shellTerminalServer.resize(this.terminalId, this.cols, this.rows);
         });
         (this.term as any).fit()
     }
 
     public async start(): Promise<void> {
         const root = await this.workspaceService.root;
-        const res = await fetch(this.endpoint.getRestUrl().toString() + '?cols=' + this.cols + '&rows=' + this.rows, {
-            method: 'POST',
-            headers: {
-                'Content-Type': 'application/json'
-            },
-            body: JSON.stringify({ uri: root.uri })
-        });
-        this.pid = await res.text();
+        this.terminalId = await this.shellTerminalServer.create(
+            <IShellTerminalServerOptions>{ rootURI: root.uri, cols: this.cols, rows: this.rows });
```

Comment:
if `shellTerminalServer` of `IShellTerminalServer` type then typecasting won't be necessary

Reply:
yep